fe/pom.xml

```diff
@@ -1025,7 +1049,7 @@ under the License.
                 <groupId>org.apache.spark</groupId>
                 <artifactId>spark-launcher_2.12</artifactId>
                 <version>${spark.version}</version>
-                <!-- <scope>provided</scope> -->
+                <scope>provided</scope>
```

Comment:
There is a CVE for the current version, and since it's an optional feature, I'd rather ignore it, and users can add this jar manually, if needed.
I will investigate the version API compatibility of Spark 2 and 3 to complete the version upgrade.